### PR TITLE
Do not include exception message when an unexpected error happens

### PR DESCRIPTION
**Pull request by mpolden**
So we don't leak anything potentially sensitive.

@jonmv

### CODE REVIEW

**Comment by mpolden** [comment]
FYI @tokle

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ public HttpResponse auditAndHandle(HttpRequest request) {
         }
         catch (RuntimeException e) {
             log.log(Level.WARNING, "Unexpected error handling '" + request.getUri() + "'", e);
-            return ErrorResponse.internalServerError(Exceptions.toMessageString(e));
+            return ErrorResponse.internalServerError("Unexpected error occurred");
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java:235]: Should we perhaps generate a UUID here, and place with both the response, and in the log, to make it easier to find relevant log entries when a user asks what happened?

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java:235]: ... or is that unnecessary?

**Comment by mpolden** [comment]
True, I can do the same other controller handlers too.

**Comment by mpolden** [comment]
> and then there are other exceptions which may leak

True, but the catch-all is likely the worst offender.

> as well as all the other handlers

Updated all handlers in the controller now. PTAL.

**Comment by jonmv** [comment]
Nice. I wonder if perhaps a server timestamp, with millis, is better than a UUID, though, since the log is ordered on time? What do you think?

**Review by jonmv — APPROVED** [review]

**Comment by mpolden** [comment]
No hard preference, but I don't see how a timestamp as ID is better? The access log and Vespa will already contain a timestamp.

**Comment by jonmv** [comment]
Yes, the point was that with a server timestamp, it'll be super-easy to find the log entry, since we'd otherwise need the approximate time of the error as auxiliary data, before searching the log for the UUID.

**Pull request merged**